### ISSUE TITLE
The number of output tensors (1) should match the number of output tensor metadata (4)

### ISSUE THREAD

**Issue by ebdjesus** [ultralytics/yolov5#4760]
Hello Guys, how you are doing?

Sorry for my lack of knowledge, but I'm trying to apply the metadata to the tflite file for use on Android and I'm getting this error.

The number of output tensors (1) should match the number of output tensor metadata (4) 

1 - Training the model

 * python3 train.py --batch-size 4 --img-size 640 --epochs 10 --data data/custom_data.yaml --weights yolov5s.pt --nosave --cache

 * custom_data.yaml

<img width="1041" alt="image" src="https://user-images.githubusercontent.com/51535131/132953025-08672a3a-adef-4646-b157-a8a7793dee3f.png">

<img width="1590" alt="image" src="https://user-images.githubusercontent.com/51535131/132952968-a536d5ca-f6fc-4b25-a075-eed930d9da80.png">

2 - Running tf.py

 * python3 models/tf.py --img 640 --weight runs/train/exp/weights/best.pt --cfg models/yolov5s.yaml --tfl-int8 --source ../train_data/images/train --ncalib 100

<img width="1577" alt="image" src="https://user-images.githubusercontent.com/51535131/132953198-9e5482cd-b2a1-49b0-afe6-b985c3f1a38b.png">

<img width="764" alt="image" src="https://user-images.githubusercontent.com/51535131/132953227-b9e53584-5019-4871-9b60-d09ff098de2a.png">

<img width="1350" alt="image" src="https://user-images.githubusercontent.com/51535131/132953253-802e7907-8271-4630-9b98-f4a5b583c42c.png">

3 - Applying metadata in .Tfflite

 * python3 metadata_writer_for_object_detection.py --model_file=model_without_metadata/final_model.tflite --label_file=model_without_metadata/labels.txt --export_directory=model_with_metadata

 * Script file: https://gist.github.com/4PixelsDev/7400df5cd6f004c4d630c849660577d6

<img width="1580" alt="image" src="https://user-images.githubusercontent.com/51535131/132953383-c53da944-158d-4d10-9bcb-8edbfdb60c0c.png">

4 - I downloaded the ssd_mobilenet_v1.tflite file from the link: https://www.tensorflow.org/lite/convert/metadata_writer_tutorial and ran the same script and the metadata file was applied

<img width="1580" alt="image" src="https://user-images.githubusercontent.com/51535131/132953697-b3339968-fd68-4353-8984-b443de4e0e96.png">

I believe I'm doing something wrong, but I haven't identified where

Thanks

**Comment by glenn-jocher** [comment]
@ebdjesus I can't speak for the 3rd party script you linked to, but the only YOLOv5 TF exports with 4 outputs are the TF.js models with agnostic NMS attached.

**Comment by ebdjesus** [comment]
Hi @glenn-jocher, thanks for your reply, I'll keep looking at the tensorflow documentation to find a better solution.

**Comment by chainyo** [comment]
Hey @ebdjesus ! Same problem here... Did you manage to find an answer to your question ?

**EDIT**: I opened an issue on the `tflite-support` repo [here](https://github.com/tensorflow/tflite-support/issues/703)

**Comment by ebdjesus** [comment]
Hi @ChainYo, I didn't solve the problem, but another guy did.
As I don't know much about the python language, I couldn't identify which file it moved.

https://github.com/ultralytics/yolov5/issues/1981

**Comment by chainyo** [comment]
@ebdjesus Thanks for the quick answer!
